### DISCUSSION TITLE
Error connecting to local sql server

### DISCUSSION THREAD

**Discussion by MnInShdw** [dbeaver/dbeaver#35876] [Q&A]
I have a local sql server Express installed on my machine (for my tests).
I can use sql server management studio to connect to this server without any problem (using both Windows and sql server authentication)
When I try to connect to this sql server with DBeaver, I receive the following error.

`The TCP/IP connection to the host HABIB, port 1342 has failed. Error: "Connection refused: no further information. Verify the connection properties. Make sure that an instance of SQL Server is running on the host and accepting TCP/IP connections at the port. Make sure that TCP connections to the port are not blocked by a firewall."
`
In sql server management studio, if I run 
_EXEC sp_readerrorlog 0, 1, N'Server is listening on'_
I receive : _Server is listening on [ 127.0.0.1 <ipv4> 1342] accept sockets 1._

The server is running because SSMS can connect to it and the port is correct.
I've tried with HABIB - localhost - HABIB\MSSQLSERVER - 127.0.0.1 ---> all returns the same error.

Can anybody help?

**Comment by TB-HunBoka** [comment]
0 experience here, just regular debugging ideas.
Steps I would try
Connect with IP address, not the host.
Disable your firewall. --> add Dbeaver to the firewall exceptions
Run Dbeaver in administrator mode --> run it as admin all the time
Ping the end point with cmd (run command prompt, type in ping 127.0.0.1) to see if the IP is restricted.

**Reply by MnInShdw** [reply]
I've done all suggested solution, still the same.
Since SSMS can connect to the server, I don't really think there's any problem with port or firewall or ip or tcp.

Any other suggestion is much appreciated.

**Comment by TB-HunBoka** [comment]
Could you check the listener controller status:
https://docs.oracle.com/en/database/oracle/oracle-database/18/xeinw/connecting-oracle-database-xe.html#:~:text=cd%20%3Coracle_home%3E%5Cbin%0Alsnrctl%20status
Check the Listening Endpoints Summary...
This summary should have the exact connection method.
